### PR TITLE
cache, boto retries

### PR DESCRIPTION
**Pull request by mwhooker**
I think caching at this layer makes sense because we can't always depend on "well-behaved clients".

### CODE REVIEW

**Comment by mwhooker** [comment]
once I get unbound running locally I'll test this

File: unbound_ec2.py [hunk 1]
```diff
@@ -110,27 +124,34 @@ def handle_forward(id, event, qstate, qdata):
...
 
     if not msg.set_return_msg(qstate):
-        qstate.ext_state[id] = MODULE_ERROR 
+        qstate.ext_state[id] = MODULE_ERROR
         return True
 
     qstate.return_msg.rep.security = 2
-    qstate.ext_state[id] = MODULE_FINISHED 
+    qstate.ext_state[id] = MODULE_FINISHED
+    if not storeQueryInCache(qstate, qstate.qinfo, qstate.return_msg, 0):
```
Comment (whilp) [inline unbound_ec2.py:153]: Cool. How are entries in this cache expired/managed? What's the 0 parameter?

Comment (mwhooker) [inline unbound_ec2.py:153]: http://unbound.net/documentation/pythonmod/modules/functions.html#storeQueryInCache

Comment (whilp) [inline unbound_ec2.py:153]: Ah, well then. ;) I'm down if we can show that queries against a cold and warm cache produce reasonable results.

**Comment by whilp** [comment]
+1 pending tests; thanks!

**Pull request closed without merging**